src/settings/global_user.rs

```diff
@@ -11,18 +12,37 @@ impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
         get_global_config()
     }
+    pub fn config_directory() -> Option<PathBuf> {
+        get_global_config_directory()
+    }
+}
+
+fn get_global_config_directory() -> Option<PathBuf> {
+    let deprecated_directory = dirs::home_dir()
+        .expect("oops no home dir")
+        .join(".wrangler")
+        .join("config");
```

Comment:
There's a slight logic issue here; user with no HOME set but XDG_* prefs will throw here.

Reply:
Yes I realized that, since it's such a niche case I was thinking it won't be worth the complexity. What do you think about returning root?

Reply:
> What do you think about returning root?
Can you please clarify?

Reply:
I mean returning '/', if there is no home directory.

Reply:
I doubt someone will configure Wrangler in `/`. XDG fallback to the home dir, I think that we should use the same logic.

Reply:
But the legacy directory will never be present in the XDG fallback right? Could you clarify what you expect?

Reply:
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html:

> If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

My understanding is that:
1. legacy config exists at `$HOME/.wrangler/config`; use it
1. XDG_* is specified; use it
1. XDG_* is empty, use fallback $HOME/.config/wrangler

Does it make sense to you?

Reply:
To clarify with point 2 expected would be $XDG_CONFIG_HOME/wrangler.
Yes that makes sense, since `dirs::config_dir` already follows the logic of 2 and 3. And first we check the legacy config the current code matches your description right?

Reply:
The unwrapping of the legacy location is a bit odd but since in the old code there was no fallback but the code would just exit. But since we still want to continue when the location is not found even when no home dir exists I am not sure how to handle this.

Reply:
Excellent (should have read the doc) and your point is right about the current resolution being broken.